docs/howtos/extending/magicgui.md

```diff
@@ -754,29 +762,63 @@ simply provide the class definition and add to the plugin manifest.
 ### `QtWidgets.QWidget`
 
 For the most control over your widget, subclass
-[`QtWidgets.QWidget`](https://doc.qt.io/qt-5/qwidget.html):
+[`QtWidgets.QWidget`](https://doc.qt.io/qt-5/qwidget.html).
+In the following example, we create a button and a dropdown list.
+The available choices for the dropdown list are the
+current layers in the viewer. For this, we use
+{func}`create_widget <magicgui.widgets.create_widget>` with the annotation
+{attr}`napari.types.ImageData`.
+Because the layer selection widget will be housed by a native `QWidget`
+and not by a `magicgui` subclass as [shown above](#parameter-annotations), we now need to
+manually connect the `reset_choices` of the created widget with the
+`viewer.layers.events` so that the available choices are synchronized
+with the current layers of the viewer:
 
 ```python
+import numpy
+import napari
+from magicgui.widgets import create_widget
 from qtpy.QtWidgets import QHBoxLayout, QPushButton, QWidget
 
-class ExampleQWidget(QWidget):
+from napari.types import ImageData
+
+
+class ExampleLayerListWidget(QWidget):
     def __init__(self, viewer: "napari.viewer.Viewer"):
         super().__init__()
         self.viewer = viewer
 
+        # Create a button
         btn = QPushButton("Click me!")
+        # Connect the click event to a function
         btn.clicked.connect(self._on_click)
 
+        # create new widget with create_widget and type annotation
+        self.layer_select = create_widget(annotation=ImageData)
+        # The `layer_select` widgets `reset_choices` method has to be connected to viewer.layers.events
+        layers_events = self.viewer.layers.events
+        layers_events.inserted.connect(self.layer_select.reset_choices)
+        layers_events.removed.connect(self.layer_select.reset_choices)
+        layers_events.reordered.connect(self.layer_select.reset_choices)
```

Comment:
```suggestion
```

Reply:
https://github.com/napari/docs/pull/68#issuecomment-2036178071

@Czaki

Reply:
We are working on nested Layers (Layer Groups). It may change a list of events. So then people will need to update their plugins if they do not depend on napari calling `reset_choices`.

Reply:
> So then people will need to update their plugins if they do not depend on napari calling reset_choices.

@Czaki I'm not happy to enshrine this behaviour in our docs right now. It's very brittle and in my opinion, quite an overreach that could lead to surprising effects on widgets totally unrelated to the layerlist. I would strongly prefer to keep the example as it currently is, because it involves no magic and no reliance on `napari` internals

Reply:
Could we explain the shortcut via `reset_choices` in the docs and also mention that there are plans that both `viewer.layers.events` and `reset_choices` might change in future, so that developers should check the corresponding source code once there plugins stop working as expected.

`reset_choices` https://github.com/napari/napari/blob/50b314f7ad2e7cd4ef21b5705b3c6b363c0857ed/napari/_qt/qt_main_window.py#L1093 shows all the relevant `layers.events` that would need to be connected individually.

`viewer.layers.events` currently is a instance subclassing https://github.com/napari/napari/blob/50b314f7ad2e7cd4ef21b5705b3c6b363c0857ed/napari/utils/events/containers/_selectable_list.py#L12

Reply:
Thanks for the offer @gatoniel —**if** we decide to take you up on it or parts of it, I think expanding on the reset_choices magic should come in a separate PR. For now I think we are still internally deciding whether we want it in the docs, as you can see 😅, so I think this should go in as-is.

Regarding the future of the event connections: in my opinion, if we start worrying in the docs about all the things we are planning to deprecate, we will never document anything. 😂 I think in general the docs should be written with regard to the current public API, unless we *really* want to discourage use of a specific API. Which is not the case here.